include/irr/asset/IMeshPacker.h

```diff
@@ -92,29 +107,55 @@ class IMeshPacker
         }
     }
 
-    virtual PackedMeshBufferData alloc(const MeshBufferType const* meshBuffer) = 0;
+
     virtual void commit(/*TODO*/) = 0;
 
 protected:
     virtual ~IMeshPacker() 
     {
-        for (int i = 0; i < 4; i++)
-            _IRR_ALIGNED_FREE(m_alctrReservedSpace[i]);
+        _IRR_ALIGNED_FREE(m_MDIDataAlctrResSpc);
+        _IRR_ALIGNED_FREE(m_idxBuffAlctrResSpc);
+        _IRR_ALIGNED_FREE(m_vtxBuffAlctrResSpc);
+        _IRR_ALIGNED_FREE(m_perInsVtxBuffAlctrResSpc);
+    }
+
+    inline size_t calcVertexSize(const SVertexInputParams& vtxInputParams, const E_VERTEX_INPUT_RATE inputRate) const
+    {
+        size_t size = 0ull;
+        for (size_t i = 0; i < SVertexInputParams::MAX_VERTEX_ATTRIB_COUNT; ++i)
+        {
+            if (vtxInputParams.enabledAttribFlags & (1u << i))
+                if(vtxInputParams.bindings[i].inputRate == inputRate)
+                    size += asset::getTexelOrBlockBytesize(static_cast<E_FORMAT>(vtxInputParams.attributes[i].format));
+        }
+
+        return size;
     }
 
 protected:
     //output mesh buffers data
     SVertexInputParams m_outVtxInputParams;
 
-    uint8_t* m_alctrReservedSpace[4];
+    void* m_MDIDataAlctrResSpc;
+    void* m_idxBuffAlctrResSpc;
+    void* m_vtxBuffAlctrResSpc;
+    void* m_perInsVtxBuffAlctrResSpc;
     core::GeneralpurposeAddressAllocator<uint32_t> m_vtxBuffAlctr;
     core::GeneralpurposeAddressAllocator<uint32_t> m_idxBuffAlctr;
     core::GeneralpurposeAddressAllocator<uint32_t> m_perInsVtxBuffAlctr;
     core::GeneralpurposeAddressAllocator<uint32_t> m_MDIDataAlctr;
 
-    const uint16_t m_maxIndexCountPerMDIData;
+    const uint16_t m_minTriangleCountPerMDIData;
+    const uint16_t m_maxTriangleCountPerMDIData;
+
+    size_t m_vtxSize;
+    size_t m_perInstVtxSize;
     
-    _IRR_STATIC_INLINE_CONSTEXPR PackedMeshBufferData invalidPackedMeshBufferData{ ~uint32_t(0), ~uint32_t(0) };
+    //TODO?:
+    //bool wasCommitCalled;
```

Comment:
move this to the implementations, you can deduce if commit was called by checking if the buffers for the data are null (not created yet)